packages/apollo-server-defer/src/__tests__/starWarsDefer-test.ts

```diff
@@ -0,0 +1,552 @@
+import { isDeferredExecutionResult } from '../execute';
+import { toArray } from 'rxjs/operators';
+
+import { StarWarsSchema } from './starWarsSchema';
+import { GraphQLError } from 'graphql';
+import { graphql } from './graphql';
+
+describe('@defer Directive tests', () => {
+  describe('Basic Queries', () => {
+    it('Can @defer on scalar types', async done => {
+      const query = `
+        query HeroNameQuery {
+          hero {
+            id
+            name @defer
+          }
+        }
+      `;
+      try {
+        const result = await graphql(StarWarsSchema, query);
+        expect(isDeferredExecutionResult(result)).toBe(true);
+        if (isDeferredExecutionResult(result)) {
+          expect(result.initialResult).toEqual({
+            data: {
+              hero: {
+                id: '2001',
```

Comment:
I thought we were going to return `name: null` here? Note that this will throw on the current Apollo Client implementation

Reply:
Fixed